File: C++/Find duplicate rows in a binary matrix.cpp
class Solution
{   
    public:
    vector<int> repeatedRows(vector<vector<int>> &matrix, int M, int N) 
    { 
        set<vector<int>>st;
        vector<int>answer;
        for(int i=0;i<M;i++)
        {
            int ele=st.size();
            st.insert(matrix[i]);
            if(st.size()==ele) answer.push_back(i);
        }
        return answer;
    } 
};

File: JAVA/Find duplicate rows in a binary matrix.java
class Solution
{
    public static ArrayList<Integer> repeatedRows(int matrix[][], int m, int n)
    {
        ArrayList<Integer> ans= new ArrayList<>();
        HashSet<String> hs= new HashSet<>();
        for(int i=0;i<m;i++){
            StringBuilder sb= new StringBuilder();
            for(int j=0;j<n;j++){
                sb.append(matrix[i][j]);
            }
           
            if(hs.contains(sb.toString()))ans.add(i);
            else hs.add(sb.toString());
        }
        return ans;
    }
}
